### DISCUSSION TITLE
Campaign is lacking and boring. The players needs motivation to go around. Here is a proposed scenario.

### DISCUSSION THREAD

**Discussion by Kaillera** [FakeFishGames/Barotrauma#4959] [Ideas and suggestions]
Scenario:

You've just been commissioned by the Europa Coalition at Dank to serve aboard a Attack class sub. The waters of Europa open up before you—explore, help expand the colonies, and make of this place what you will.

Your grandma "The Gay" has written a letter to you promising that you can finally inherit your dead father's 50,000 Marks. She tells you to visit her if you want it.

You go to your Grandma and she tells you in order to claim the inheritance, you'll need to go to your Uncle "Dank Gay" in Kimju. However area beyond Zone 1 has been blocked by emerging Alien structures that requires some kind of Key.

Grandma Intel 1: She tells you that the key can be found between **Natural Formation connected to Natural Formation**

Intel 2: You go to Apep and heard that the **Alien Key is near Hecta.**

Intel 3: You go to Tara and heard that the **Alien Key is between Mori and Hecta**

![image](https://user-images.githubusercontent.com/5086682/106489163-7831fc80-6482-11eb-90a9-4328ca996c06.png)

===========================

The problem with Campaign in Barotrauma currently is the lack of reasons to do anything in Campaign. Sure going to the far right to reach The End is one, but it's shallow in that you only need to experience it once. Driving right all the time gets pretty dull. Players can stay pretty much at the beginning of the map and still have the same experience going all the way to the right. There is no reasons to actually go to any particular places either. I don't believe the shrinking map hazard will change the current gameplay problem we currently have. The players will just move right when they have to move.

I think the player should get quests that leads the players to go into certain direction. In my above example, the player will be offered 50,000 marks for completing it. They start their adventure by going to Grandma. Grandma tells them that to get it, they have to get to their Uncle.

The zone where Uncle resides in is sectioned off and has to be unlocked by obtaining a Alien Key. The player won't know where the Alien Key is, so either they get lucky or get drip fed the intel. We will go with 3 stages of intel. No matter where they get their intel from, they will get a vague hint, a better hint, and a decisive hint. This creates a adventure for players to have to visit all sorts of map tiles. They can get lucky or gamble with lacking intel, or they can visit certain places and hope to get the rest of the intel.

Whether the player get lucky or got the decisive intel, the player will now have to adventure to the specific place to get the Alien Key. This key can be hidden inside a ruin which nowadays all the players skip because there is little reasons to actually enter one. What if we promise them that they can find a key now? Of course they can still ignore most of the ruins and wait for the decisive intel, but even those intel aren't likely to be found in every outposts.

With the vague intel above, the player has 8 tiles to check out. With the better intel above, they managed to knock down the possible locations to 3. Even if they didn't get the decisive intel, there is now reasons to go to places, and that'll add meaning to the world map we currently have now. If the scenario played out like it did, I would've have to driven in this manner:

![Capture](https://user-images.githubusercontent.com/5086682/106492462-f04df180-6485-11eb-81bd-232c1a2a9d36.PNG)

Now the player can move forward after obtaining the Alien Key. We can pretty much inject the same gameplay loop or different variation of it with some sort of lure to reach The End.

**Comment by RadCatish** [comment]
I feel like this is a little harsh to the devs, but I do broadly agree. We're enjoying our first playthrough of the campaign, but I think that replayability could definitely be improved if the format of the campaign was changed up a bit. We've done a bit of toing and froing to clear breeding grounds and try and help new outposts be set up, and it's very nice to feel that our actions are influencing the progression of the world. I hope that the update which will expand on factions will also give a lot more motivation and intrigue to Europa, and more reason to move around the map, more investment in the fate and development of outposts and colonies, and more variety in playthroughs

**Comment by Kulik242** [comment]
The campaign part is probably among the things that they will focuss on after they have most of the mechanics and content. No idea whether they go for a privateer style sandbox, or objective driven gameplay and if so how much of it will be scripted and how much emergent.

One thing to consider is that if they stick to the multiplayer oriented concept, the campaign is too long. I play with my group and on average, we do like 2-4 missions per week. Seeing how much progress we made in one month that we had been playing, it will take months to finish. I personally wouldn't mind a shortened version as an option, kind of alike FTL' s "Dash to right".

**Reply by Kaillera** [reply]
> The campaign part is probably among the things that they will focuss on after they have most of the mechanics and content. No idea whether they go for a privateer style sandbox, or objective driven gameplay and if so how much of it will be scripted and how much emergent.
> 
> One thing to consider is that if they stick to the multiplayer oriented concept, the campaign is too long. I play with my group and on average, we do like 2-4 missions per week. Seeing how much progress we made in one month that we had been playing, it will take months to finish. I personally wouldn't mind a shortened version as an option, kind of alike FTL' s "Dash to right".

FTL's Dash to the Right style is actually coming to next version of the game. The map on the left will shrink soon. Problem is most of the experienced players are already dashing to the right even without anything telling them to do so.

**Comment by Xeavas** [comment]
I would agree on having more side quests and quest you can take on as an option, but not for locking you out of the zone requiring you to do a certain quest to unlock said next zone, to some it'll seem like padding the game's playtime out. 

Furthermore, I would like to express a wish for more openworld type availability focusing on pulling the player than pushing, interesting things and branching missions to lure the player in, event storytelling that can lead to a mission, multi objective missions too. To enhance them the possibility of specialized tier 4 outpost would offer unique upgrades/implants/rewards needing you to trek there if you wish to do so, however the primary focus is that it should be **optional** but not a requirement, relying on hooking the player with good worldbuilding and mission variety 

Lastly, I would like to show objection to the "FTL like map" (currently), I would prefer a more circular map but with the current map artstyle, it would be easier to illustrate the "crater" and depth, and making it more open to "sideways" navigation rather than heading to the "right" (or centre in the circular map format), as for the radiation or something related for the "shrinking" perhaps it could be complemented with a fleshed out new game + allowing you to play once irradiated/shrinked regions (like you defeated the evil or something that caused the shinking) turning it to a more open world like experience where you can go through things you might've ignored.

**Comment by ezjamsen** [comment]
Thanks for the feedback and the ideas, it's true the campaign is a bit unsophisticated right now, and we'll see what makes sense, short, medium and long term.

**Comment by RadCatish** [comment]
Perhaps two different campaigns? One long-haul which the aim is to establish a thriving human colony on Europa and involves lots of going backwards and forwards, ferrying supplies and people, helping out settlements/outposts you've been to previously, letting you get involved in the politics (and overthrow the Europa Coalition!), with near-endless development of the map, chances to upgrade your sub, small intrigues and impactful decisions. And a second, run-to-the-deeps and kill the Endworm as fast as possible, all guns blazing type campaign which focuses on your sub and crew and battling monsters? Much like what we currently have, but a bit shorter.

**Comment by Kulik242** [comment]
Motivation to move faster in the campaign could be that the endworm would lie eggs that would hatch. The hatchlings would mature and spread causing trouble on the campaign map. If you linger, the infestation would get worst eventually killing outposts from right to left.  
Moving through infested portion of map would be problematic cause without outposts there would be no possibility to resupply and the hatchling worms attacks would constantly drain ammo and welding fuel in addition to fuel bars.
The campaign difficulty would controll how fast the infestation starts and how fast it spreads. 
As an option to diving into the infested territory and killing the nest, the players could push back the infestation by doing missions to re-establish outposts. As the constant worm shooting may get boring, these missions could be more engineering oriented like building some sort of repulsor installations and fixing abandoned and infested outposts.
Caves and alien ruins may play a role here. The caves could contain a unique plant that can be used to craft a worm poison. The aliens, which may gotten overrun by previous worm infestation and evacuated, could have artifacts that repulse worms and are needed to build the repulsors.

**Reply by Kulik242** [reply]
....OR.... The infestation could start on the left, in the cold caverns!!! You would have to "escape" to the right untill you research the poison and repulsor technology. And then regain the infested lands. :)

**Reply by RadCatish** [reply]
I love both of these ideas, this would definitely add more meaning to the campaign!

**Comment by NilanthAnimosus** [comment]
I do think campaign needs more substance and well, some kind of story/narrative and global mission/thing to do, I am not sure about this initial concept specifically, I feel it would be unusual places on different ends of the map knowing Intel on some keys location or inheritances and such would just be odd overall.

It would make more sense in my eyes to implement and to have hearsay/key people in the game update some form of global (not mission-specific) mission(s) and/or have a different take on it, such as you are tasked with breaking a barrier that has formed that researchers believe say - has something to do with the alien structures due to some unnatural phenomenon. Your submarine could be tasked by the coalition (or any other in-game company/entity that exists, even renegades) with learning more about this problem that has cut off other colonies.

This narrative could still include the Intel concept, except you could be chasing down clues from station dwellers from what other submarines have learned, then say learn a particular submarine had some form of key or cache they were transporting but went missing on route to X from Y, which going the right direction may indicate if they passed by a particular station from talking to those inside the outpost.

Finally, as with the other ideas I like the idea of a semi time-limit effecting how many stations exist and general station expansion/loss on not getting the other side open, Be it resources, starvation, or a greater threat that needs to be neutralized.

Upon gaining access to the next area (Say, you have found this particular wreck and transported said 'strange' key/artifact to the relevant outpost that can research and tell you how its used or where to find out how to use it, then done that task) There could be yet another revealed threat learned from on board the outposts. IE. an end boss, or end location that nobody has ever reached but is causing significant, perhaps expanding problems.

Be it from the left an ever-expanding biological/creature/flora issue, or from the right some colossal creature (IE. end worm or strange facility) and/or its spawn causing widespread devastation from an unseen threat which decreases crush depth denying approach without submarine upgrades (Again, these are mechanics we have or could improve how well they mesh into other things). could even be both if you want a safe and difficult area to shrink simultaneously. Of course the game can be open ended after finishing all story elements, but give something more to do in a broad sense of things.

If one wanted to be crazy one could straight up mount and blade it with factions and influencing stations and some form of station defense or attack and such, but I feel that is so far beyond what seems feasible to implement today without it being almost crazy (However awesome or bad it would be to turn this into a complete sandbox-like). I feel all of that though would be separate features or discussions and 'add to' a campaign scene however.

Finally, it could simply be made an option to skip part, or all of the story, to allow for those shorter (or just sandbox-like) games. so you have your long haul, your short haul, and your basically dipping your feet in.

**Comment by HeatedWafflez** [comment]
I can definitely agree with this, the campaign will definitely need to be built upon. The fact that it's so long means that some sort of story will need to be put into place to give the players a bigger incentive to learn about Europa and its secrets. Talking to NPC's should potentially reveal important locations to the story and perhaps there could even be multiple endings depending on the actions you take. I believe a potential first scenario could be that the separatists and coalition are in a conflict that you can choose to take sides on. Depending on who wins the conflict your chosen victorious faction may choose to investigate the center. You could also lose the conflict and be forced to leave to the center since the victorious faction labels you as traitors and as a result you'll be attacked on sight. This also imposes a sort of FTL scenario because of the fact they'll be establishing colonies and you'll be forced to be on the run. 

 These are just ideas though and I have no idea where the devs plan to go with the campaign. One thing for certain though is that it will need to see change some time in the future.
